types/application.ts

```diff
@@ -465,3 +465,51 @@ export interface UploadDocumentRequest {
     fileType: "jpeg" | "png" | "pdf"
 }
 
+export interface CreateApplicationFormRequest{
+    "type": "applicationForm"
+    "attributes": {
+        /**
+         * See [Tags](https://developers.unit.co/#tags). Tags that will be copied to the customer that this application creates(see [Tag Inheritance](https://developers.unit.co/#tag-inheritance)).
+         */
+        tags?: object
+        /**
+         * Optional. Add data that is already known about the end-customer to be auto populated on the form.
+         */
+        applicantDetails?: ApplicationFormPrefill
+    }
+}
+
+export interface CreateApplicationFormResponse{
+    "type": "applicationForm"
+    "id": string
+    "attributes": {
+        /**
+         * See [Tags](https://developers.unit.co/#tags). Tags that will be copied to the customer that this application creates(see [Tag Inheritance](https://developers.unit.co/#tag-inheritance)).
+         */
+        tags?: object
+        /**
+         * The URL of the application form for the end-customer to access
+         */
+        url: string
+        /**
+         * Data that is already known about the end-customer to be auto populated on the form.
+         */
+        applicantDetails: ApplicationFormPrefill
+    }
+}
+
+export type ApplicationFormPrefill = IndividualApplicationFormPrefill | BusinessApplicationFormPrefill
+
+export type IndividualApplicationFormPrefill =  {
+    /**
+     * Optional. One of "Individual" or "SoleProprietorship".
+     */
+    applicationType: "Individual" | "SoleProprietorship"
+} & Omit<CreateIndividualApplicationRequest["attributes"], "tags" | "idempotencyKey">
+
+export type BusinessApplicationFormPrefill =  {
+    /**
+     * Optional. One of "Business".
+     */
+    applicationType: "Business"
+} & Omit<CreateBusinessApplicationRequest["attributes"], "tags" | "idempotencyKey">
```

Comment:
Nice, but move to types/applicationForm.ts

Reply:
i move everything into his own new type ApplicationFormPrefill, that behave as one large "bag" of flatten optionals fields. Let me know if any other comment appears. Thank you on your quick response.